src/CaptureFile/CaptureFileTest.cpp

```diff
@@ -0,0 +1,125 @@
+  orbit_grpc_protos::ClientCaptureEvent event2 =
+      CreateInternedStringCaptureEvent(kNotAnAnswerKey, kNotAnAnswerString);
+
+  auto write_result = output_stream->WriteCaptureEvent(event1);
+  ASSERT_FALSE(write_result.has_error()) << write_result.error().message();
+  write_result = output_stream->WriteCaptureEvent(event2);
+  ASSERT_FALSE(write_result.has_error()) << write_result.error().message();
+  auto close_result = output_stream->Close();
+  ASSERT_FALSE(close_result.has_error()) << close_result.error().message();
+
+  auto capture_file_or_error = CaptureFile::OpenForReadWrite(temporary_file.file_path());
+  ASSERT_TRUE(capture_file_or_error.has_value()) << capture_file_or_error.error().message();
+  std::unique_ptr<CaptureFile> capture_file = std::move(capture_file_or_error.value());
+  EXPECT_EQ(capture_file->GetSectionList().size(), 0);
+
+  auto section_number_or_error = capture_file->AddSection(1, 333);
+  ASSERT_TRUE(section_number_or_error.has_value()) << section_number_or_error.error().message();
+  ASSERT_EQ(capture_file->GetSectionList().size(), 1);
+  EXPECT_EQ(section_number_or_error.value(), 0);
+  {
+    const auto& capture_file_section = capture_file->GetSectionList()[0];
+    EXPECT_EQ(capture_file_section.size, 333);
+    EXPECT_EQ(capture_file_section.type, 1);
+  }
+
+  // Write something to the section
+  std::string something{"this is something"};
+  constexpr uint64_t kOffsetInSection = 25;
+  auto write_to_section_result =
+      capture_file->WriteToSection(0, kOffsetInSection, something.c_str(), something.size());
+  ASSERT_FALSE(write_to_section_result.has_error()) << write_to_section_result.error().message();
+
+  {
+    std::string content;
+    content.resize(something.size());
+    auto read_result =
+        capture_file->ReadFromSection(0, kOffsetInSection, content.data(), something.size());
+    ASSERT_FALSE(read_result.has_error()) << read_result.error().message();
+    EXPECT_EQ(content, something);
+  }
+
+  // Reopen the file to make sure this information was saved
+  capture_file.reset();
+
+  capture_file_or_error = CaptureFile::OpenForReadWrite(temporary_file.file_path());
+  ASSERT_TRUE(capture_file_or_error.has_value()) << capture_file_or_error.error().message();
+  capture_file = std::move(capture_file_or_error.value());
+  EXPECT_EQ(capture_file->GetSectionList().size(), 1);
+  {
+    const auto& capture_file_section = capture_file->GetSectionList()[0];
+    EXPECT_EQ(capture_file_section.type, 1);
+    EXPECT_EQ(capture_file_section.size, 333);
+    EXPECT_GT(capture_file_section.offset, 0);
+  }
+
+  {
+    std::string content;
+    content.resize(something.size());
+    auto read_result =
+        capture_file->ReadFromSection(0, kOffsetInSection, content.data(), something.size());
+    ASSERT_FALSE(read_result.has_error()) << read_result.error().message();
+    EXPECT_EQ(content, something);
+  }
+}
+
+TEST(CaptureFile, OpenInvalidCaptureFile) {
```

Comment:
There are couple of reasons for a file being invalid, right? Should we have a test for each reason?